### PR TITLE
Added support for Inverse references

### PR DESCRIPTION
**Pull request by destogl**
Working on #161 I found out that inverse references are missing (from child to parent). So here is initial implementation. The result is cross-compared with server based on Unified Automation SDK and Textual specification (Part 3) is consulted. Following major changes are done:

* Hierarchical references are automatically added (parent <--> child)
* create_method now returns Node (as all other create methods (variable, object))
* create_folder is now merged with create_object since the difference is in only one variable
* On multiple places are written constants instead of hard-coded numeric values
* Tests for references are written and some others are extended

### CODE REVIEW

File: opcua/server/address_space.py [hunk 1]
```diff
@@ -108,7 +108,8 @@ def _suitable_reftype(self, ref1, ref2, subtypes):
         """
         if ref1.Identifier == ref2.Identifier:
             return True
-        if not subtypes and ref2.Identifier == ua.ObjectIds.HasSubtype:
+        #TODO: Please check the changes here
+        elif not subtypes:
```
Comment (destogl) [inline opcua/server/address_space.py]: @oroulet Please check this line. i change it to get all test to work. I don't understand it at all, for me the if statement before doesn't make any sense :/

Comment (oroulet) [inline opcua/server/address_space.py]: did the original implementation break something? This is copied from C++ and I kind of remember we had issues here. But removing the last test as you did seems to make sense... if it does not break tests we can try with your version...

File: tests/tests_common.py [hunk 2]
```diff
@@ -856,14 +882,57 @@ def test_method_array2(self):
...
+        self.assertEqual(v.get_type_definition(), ua.ObjectIds.BaseDataVariableType)
+
+        p = o.add_property(3, 'MyProperty', 2)
+        nodes = o.get_referenced_nodes(refs=ua.ObjectIds.HasProperty, direction=ua.BrowseDirection.Forward, includesubtypes=False)
+        self.assertTrue(p in nodes)
+        nodes = p.get_referenced_nodes(refs=ua.ObjectIds.HasProperty, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+        self.assertTrue(o in nodes)
+        self.assertEqual(p.get_parent(), o)
+        self.assertEqual(p.get_type_definition(), ua.ObjectIds.PropertyType)
+
```
Comment (oroulet) [inline tests/tests_common.py]: you added tests! good reason to trust you new code :-)

**Comment by oroulet** [comment]
looks all good

File: tests/tests_common.py (hunk 2 above)
Comment (destogl) [inline tests/tests_common.py]: Yes I decided for the future to try to use Test-driven development, since it is easier to get what you want :)

**Pull request merged**